djstripe/event_handlers.py

```diff
@@ -14,95 +15,195 @@
 
 """
 
+from django.utils import six
 from . import webhooks
 from .models import Charge, Customer, Card, Subscription, Plan, Transfer, Invoice, InvoiceItem
 
-STRIPE_CRUD_EVENTS = ["created", "updated", "deleted"]
 
-
-# ---------------------------
-# Charge model events
-# ---------------------------
-@webhooks.handler(['charge'])
-def charge_webhook_handler(event, event_data, event_type, event_subtype):
-    versioned_charge_data = Charge(stripe_id=event_data["object"]["id"]).api_retrieve()
-    Charge.sync_from_stripe_data(versioned_charge_data)
-
-
-# ---------------------------
-# Customer model events
-# ---------------------------
 @webhooks.handler_all
 def customer_event_attach(event, event_data, event_type, event_subtype):
+    """ Makes the related customer available on the event for all handlers. """
+    event.customer = None
+    crud_type = CrudType.determine(event_subtype, exact=True)
 
-    if event_type == "customer" and event_subtype in STRIPE_CRUD_EVENTS:
-        stripe_customer_id = event_data["object"]["id"]
+    if event_type == "customer" and crud_type.valid:
+        customer_stripe_id = event_data["object"]["id"]
     else:
-        stripe_customer_id = event_data["object"].get("customer", None)
+        customer_stripe_id = event_data["object"].get("customer", None)
 
-    if stripe_customer_id:
-        try:
-            event.customer = Customer.objects.get(stripe_id=stripe_customer_id)
-        except Customer.DoesNotExist:
-            pass
+    if not customer_stripe_id:
```

Comment:
We're typically less keen on branch logic because of the nesting and usually because it forces the code to be longer in terms of line length, *but only if adhering to E501 in PEP-8*.  If line length isn't an issue I can see why it might not be as undesirable. ;-)  I'll look into reverting to branch logic for these now.